contracts/sol6/Dao/EmergencyFeeHandler.sol

```diff
@@ -0,0 +1,201 @@
+    /// @param platformFeeWei Fee amount in wei the platfrom wallet is entitled to.
+    function handleFees(
+        address[] calldata rebateWallets,
+        uint256[] calldata rebateBpsPerWallet,
+        address platformWallet,
+        uint256 platformFeeWei
+    ) external payable override onlyKyberNetwork {
+        // handle platform fee
+        feePerPlatformWallet[platformWallet] = feePerPlatformWallet[platformWallet].add(
+            platformFeeWei
+        );
+
+        uint256 feeBRRWei = msg.value.sub(platformFeeWei);
+        totalPlatformFeeWei = totalPlatformFeeWei.add(platformFeeWei);
+        if (feeBRRWei == 0) {
+            // only platform fee paid
+            emit FeeDistributed(
+                platformWallet,
+                platformFeeWei,
+                0,
+                0,
+                rebateWallets,
+                rebateBpsPerWallet,
+                0
+            );
+            return;
+        }
+
+        (bool success, ) = address(this).call(
+            abi.encodeWithSignature(
+                "calculateAndRecordFeeData(address[],uint256[],uint256,address,uint256)",
+                rebateWallets,
+                rebateBpsPerWallet,
+                feeBRRWei,
+                platformWallet,
+                platformFeeWei
+            )
+        );
+        if (!success) {
+            emit HandleFeeFailed(rebateWallets, rebateBpsPerWallet, feeBRRWei);
+        }
+    }
+
+    function calculateAndRecordFeeData(
+        address[] calldata rebateWallets,
+        uint256[] calldata rebateBpsPerWallet,
+        uint256 feeBRRWei,
+        address platformWallet,
+        uint256 platformFeeWei
+    ) external {
+        require(msg.sender == address(this), "only Feehandler contract can call this function");
+        uint256 rebateWei = feeBRRWei.mul(rebateBps).div(BPS);
+        uint256 rewardWei = feeBRRWei.mul(rewardBps).div(BPS);
+
+        rebateWei = updateRebateValues(rebateWei, rebateWallets, rebateBpsPerWallet);
+
+        totalRewardWei = totalRewardWei.add(rewardWei);
+
+        emit FeeDistributed(
+            platformWallet,
+            platformFeeWei,
+            rewardWei,
+            rebateWei,
+            rebateWallets,
+            rebateBpsPerWallet,
+            feeBRRWei
+        );
+    }
+
+    /// @dev claim accumulated fee per platform wallet. Called by any address
+    /// @param platformWallet the wallet to claim fee for. Total accumulated fee sent to this wallet.
+    /// @return feeWei amount of fee claimed
+    function claimPlatformFee(address platformWallet) external returns (uint256 feeWei) {
+        require(feePerPlatformWallet[platformWallet] > 1, "no fee to claim");
+        // Get total amount of fees accumulated
+        uint256 amount = feePerPlatformWallet[platformWallet].sub(1);
+
+        // redundant check, but better revert message
+        require(totalPlatformFeeWei >= amount, "platform fee amount too high");
```

Comment:
change to assert.
Since theoretically can't happen

Reply:
change as request